src/transformers/modeling_reformer.py

```diff
@@ -600,26 +722,37 @@ def _attend(
         sorted_bucket_idx_per_hash,
         attention_mask,
         head_mask,
-        sequence_length,
+        do_standard_self_attention,
+        do_cached_attention,
     ):
-
         # look at previous and following chunks if chunked attention
-        if self.chunk_length < sequence_length:
+        if not do_standard_self_attention:
             key_vectors = self._look_adjacent(key_vectors, self.num_chunks_before, self.num_chunks_after)
             value_vectors = self._look_adjacent(value_vectors, self.num_chunks_before, self.num_chunks_after)
 
         # get logits and dots
+        # (BS, NumAttn, NumHash x NumChunk, Chunk_L x Hidden),(BS, NumAttn, NumHash x NumChunk, Chunk_L * (Num_bef + Num_aft + 1) x Hidden) -> (BS, NumAttn, NumHash x NumChunk, Chunk_L, Chunk_L * (1 + Num_bef + Num_aft))
         query_key_dots = torch.matmul(query_vectors, key_vectors.transpose(-1, -2))
 
         # free memory
         del query_vectors, key_vectors
 
         # if chunked attention split bucket idxs to query and key
-        if self.chunk_length < sequence_length:
+        if not do_standard_self_attention:
```

Comment:
do we hit all these cases? Do we need to?

Reply:
Yes lot's of weird special cases for Reformer caching